File: backend/src/main/java/com/kaamchahiye/app/service/AuthService.java
package com.kaamchahiye.app.service;

import com.kaamchahiye.app.dto.request.LoginRequest;
import com.kaamchahiye.app.dto.request.RegisterRequest;
import com.kaamchahiye.app.dto.response.AuthResponse;
import com.kaamchahiye.app.dto.response.UserDto;

public interface AuthService {
    AuthResponse login(LoginRequest request);
    AuthResponse register(RegisterRequest request);
    UserDto getCurrentUser(String phoneNumber);
}

File: frontend/src/services/authService.js
import apiClient from './apiClient';

export const authService = {
  login: async (phoneNumber, password) => {
    const response = await apiClient.post('/auth/login', { phoneNumber, password });
    if (response.data.token) {
      localStorage.setItem('jwt_token', response.data.token);
      localStorage.setItem('user_info', JSON.stringify(response.data.user));
    }
    return response.data;
  },

  register: async (userData) => {
    const response = await apiClient.post('/auth/register', userData);
    if (response.data.token) {
      localStorage.setItem('jwt_token', response.data.token);
      localStorage.setItem('user_info', JSON.stringify(response.data.user));
    }
    return response.data;
  },

  getCurrentUser: async () => {
    const response = await apiClient.get('/auth/me');
    return response.data;
  },

  logout: () => {
    localStorage.removeItem('jwt_token');
    localStorage.removeItem('user_info');
  }
};

export default authService;
